plugins/mcreator-core/blockly/js/extensions_procedure_deps.js

```diff
@@ -0,0 +1,225 @@
+Blockly.Extensions.register('procedure_dependencies_tooltip',
+    function () {
+        let thisBlock = this;
+        this.setTooltip(function () {
+            const depList = javabridge.getAllDependencies(thisBlock.getFieldValue('procedure'));
+            if (depList.length == 0)
+                return javabridge.t('blockly.extension.procedure_dep_tooltip.empty');
+            let tooltip = javabridge.t('blockly.extension.procedure_dep_tooltip');
+            for (const dependency of depList)
+                tooltip += '\n' + dependency;
+            return tooltip;
+        });
+    });
+
+Blockly.Extensions.register('procedure_dependencies_onchange_mixin',
+    function () {
+        this.setOnChange(function (changeEvent) {
+            // Trigger the change only if a block is changed, moved, deleted or created
+            if ((changeEvent.type !== Blockly.Events.BLOCK_CHANGE ||
+                changeEvent.element !== 'field') &&
+                changeEvent.type !== Blockly.Events.BLOCK_CREATE) {
+                return;
+            }
+            const group = Blockly.Events.getGroup();
+            // Makes it so the block change and the unplug event get undone together.
+            Blockly.Events.setGroup(changeEvent.group);
+            let valid = true;
+            const procedure = this.getFieldValue('procedure');
+            for (var i = 0; this.getField('name' + i); i++) {
+                const prevType = this.getInput('arg' + i).connection.getCheck();
+                // Set input checks from dependency type
+                this.getInput('arg' + i).setCheck(javabridge.getDependencyType(
+                    this.getFieldValue('procedure'), this.getFieldValue('name' + i)));
+                const newType = this.getInput('arg' + i).connection.getCheck();
+                // Disable the block if procedure is defined and invalid used dependencies are present
+                if (javabridge.hasDependency(procedure, this.getFieldValue('name' + i)) &&
+                    (!procedure || procedure !== '') && !newType && this.getInputTargetBlock('arg' + i) &&

```

Comment:
The `(!procedure || procedure !== '')` part always returns true

Reply:
:heavy_check_mark: Fixed